packages/terra-date-input/src/DateInput.jsx

```diff
@@ -738,21 +787,28 @@ class DateInput extends React.Component {
 
     // Using the state of month, day, and year to create a formatted date value
     let dateValue = '';
+    let completeDateValue;
 
     if (month.length > 0 || day.length > 0 || year.length > 0) {
       dateValue = `${year}-${month}-${day}`;
     }
 
-    this.formatDescriptionId = `terra-date-input-description-format-${this.uuid}`;
+    if (dateValue.length === 'xx-xx-xxxx'.length) {
+      completeDateValue = dateValue;
+    }
 
     const format = DateInputUtil.getDateFormat(this.props);
+    const label = a11yLabel || this.props.intl.formatMessage({ id: 'Terra.date.input.labelDefault' });
 
     return (
       <div
         {...customProps}
         className={dateInputClassNames}
         ref={(element) => { this.dateInputContainer.current = element; if (refCallback) { refCallback(element); } }}
+        role={isA11yControlled ? undefined : 'group'}
```

Comment:
I used `undefined` because that is the default value of role in HTML.
```
> let div = document.createElement('div')
undefined
> div.role
undefined
```

Reply:
<img width="584" alt="image" src="https://user-images.githubusercontent.com/9022825/156026764-35189152-014e-4138-ad21-33130a46d9ea.png">
The default value of any attribute that doesn't have a value is `null`. This mixup illustrates the point pretty well: with `null` you know you're getting a value; it's just a non-value value. `undefined` just means there's no defined value for the thing you're trying to access. 
In React, even if you pass `undefined` as the value of an element's attribute, it will still render an element with a `null` value for that attribute.

Reply:
TIL!